service/status-report/main.go

```diff
@@ -117,9 +118,12 @@ func main() {
 	setOperatorInformation(ctx, cl, status, operatorNamespace)
 	setClusterInformation(ctx, cl, status)
 	setStorageQuotaUtilizationRatio(ctx, cl, status)
-	statusResponse, err := providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status)
-	if err != nil {
-		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
+	statusResponse := &providerpb.ReportStatusResponse{}
+	if storageClient.GetDeletionTimestamp().IsZero() {
+		statusResponse, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status)
+		if err != nil {
+			klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
+		}
```

Comment:
Why does this change make sense? I would think that if we are stuck on deletion we should still report heartbeat info. The reporting will stop the min that the job will be gone which should happen as part of the clean up we should do when the client CR is getting deleted

Reply:
Agreed, 
changing it to returning error on Offboarding failure